src/rpcmining.cpp

```diff
@@ -766,3 +766,27 @@ Value estimatepriority(const Array& params, bool fHelp)
 
     return mempool.estimatePriority(nBlocks);
 }
+
+Value getblocksubsidy(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+            "getblocksubsidy index\n"
+            "\nReturns block subsidy reward, taking into account the mining slow start, of block at index provided.\n"
+            "\nArguments:\n"
+            "1. index         (numeric, required) The block index\n"
+            "\nResult:\n"
+            "amount           (numeric) The block reward amount in ZEC.\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getblocksubsidy", "1000")
+            + HelpExampleRpc("getblockubsidy", "1000")
+        );
+
+    LOCK(cs_main);
```

Comment:
I moved the lock to just before chainActive.tip() to match existing (i.e. upstream) usage.